### ISSUE TITLE
docker load cannot load xz archive files

### ISSUE THREAD

**Issue by mbengit** [docker/for-mac#5271]
Executing `docker load < archive.xz` on the command line produces this error: 

`exec: "xz": executable file not found in $PATH`

This issue was introduced in **Docker Desktop 3.1.0** and is easily reproducible on MacOS and Windows.

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 63F376B3-ACF5-425B-843D-AAED828D01E7/20210119170855

**Comment by mbengit** [comment]
It seems likely that the new Debian container added in 3.1.0 does not support `xz`.

**Comment by stephen-turner** [comment]
Yes, it could be that. @simonferquel PTAL.

**Comment by mbengit** [comment]
Hi, is there any progress on a fix for this issue?
I work on a product which depends on being able to load an xz-archived image to the local docker registry, so this issue has a sizeable impact.

**Comment by mat007** [comment]
Hi, the fix will be available in the next release.

**Comment by stephen-turner** [comment]
Yes, we're working on it. It should be in the next version.

**Comment by qalinn** [comment]
Hi @mat007, 

Could you tell me when is planned the next release? 
I am also impacted by this issue.

Thank you!

**Comment by stephen-turner** [comment]
Sorry, we don't have a date yet.

**Comment by qalinn** [comment]
@stephen-turner That means days, weeks, months? Or how does it release a new version?

**Comment by stephen-turner** [comment]
Definitely weeks, not days or months.

**Comment by qalinn** [comment]
@stephen-turner  Thank you

**Comment by Djelibeybi** [comment]
As a workaround, you can use `buildx` and the default `moby/buildkit:buildx-stable-1` image which appears to still hve `xz` in it, i.e.:

```
$ docker buildx create --driver docker-container --name fix-xz --use
$ docker buildx build --load -t image:tag .
[+] Building 30.5s (7/7) FINISHED
 => [internal] booting buildkit                                                                                                                                              12.7s
 => => pulling image moby/buildkit:buildx-stable-1                                                                                                                           11.9s
 => => creating container buildx_buildkit_fix-xz0                                                                                                                          0.8s
 => [internal] load build definition from Dockerfile                                                                                                                          0.0s
 => => transferring dockerfile: 174B                                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                               0.0s
 => [internal] load build context                                                                                                                                             1.6s
 => => transferring context: 24.40MB                                                                                                                                          1.6s
 => [1/1] ADD archive.tar.xz /                                                                                                                        3.6s
 => exporting to oci image format                                                                                                                                            12.2s
 => => exporting layers                                                                                                                                                       6.5s
 => => exporting manifest sha256:2f98ad89fba24c0424199b31a9bdfe9e68507da99747404750eff5dd7702036c                                                                             0.0s
 => => exporting config sha256:156d3cd0610c9bda95c92c50dbb72fd9bfded8596521fc99f91111e7b06af58c                                                                               0.0s
 => => sending tarball                                                                                                                                                        5.7s
 => importing to docker

**Comment by stephen-turner** [comment]
This is fixed in versions 3.2.0 and 3.2.1. Closing the ticket now.
